src/librustc_typeck/astconv.rs

```diff
@@ -480,6 +496,7 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
         parent_substs: &[subst::GenericArg<'tcx>],
         has_self: bool,
         self_ty: Option<Ty<'tcx>>,
+        arg_count_mismatch: bool,
```

Comment:
Hmm, maybe you can replace this with a `Result<(), GenericArgCountMismatch>`?
Especially if it's the same `bool` returned by some functions.